src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -28,642 +37,878 @@ public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoad
+                loader.AddDependencyLocation(testFixture.Beta.Path);
+                Assembly beta = loader.LoadFromPath(testFixture.Beta.Path);
+
+                var b = beta.CreateInstance("Beta.B")!;
+                var writeMethod = b.GetType().GetMethod("Write")!;
+                var exception = Assert.Throws<TargetInvocationException>(
+                    () => writeMethod.Invoke(b, new object[] { sb, "Test B" }));
+                Assert.IsAssignableFrom<FileNotFoundException>(exception.InnerException);
+
+                var actual = sb.ToString();
+                Assert.Equal(@"", actual);
+            });
         }
 
         private static void VerifyAssemblies(IEnumerable<Assembly> assemblies, params (string simpleName, string version, string path)[] expected)
         {
             Assert.Equal(expected, Roslyn.Utilities.EnumerableExtensions.Order(assemblies.Select(assembly => (assembly.GetName().Name!, assembly.GetName().Version!.ToString(), assembly.Location))));
         }
 
-        [ConditionalFact(typeof(CoreClrOnly))]
+        /// <summary>
+        /// Verify the set of asesmblies loaded as analyzer dependencies are the specified assembly paths
+        /// </summary>
+        private static void VerifyDependencyAssemblies(DefaultAnalyzerAssemblyLoader loader, params string[] assemblyPaths)
+        {
+            IEnumerable<Assembly> loadedAssemblies;
+
+#if NETCOREAPP
+            // This verify only works where there is a single load context.
+            var alcs = DefaultAnalyzerAssemblyLoader.TestAccessor.GetOrderedLoadContexts(loader);
+            Assert.Equal(1, alcs.Length);
+
+            loadedAssemblies = alcs[0].Assemblies;
+#else
+
+            // Unfortunately in .NET Framework we cannot determine which Assemblies are in the LoadFrom context
+            // which is what we want here. Our ability to verify here is less robust and instead we filter out 
+            // all the DLLs we know are simply a part of the application and examine what is left.
+            loadedAssemblies = AppDomain.CurrentDomain
+                .GetAssemblies()
+                .Where(x => isInLoadFromContext(loader, x));
+
+            static bool isInLoadFromContext(DefaultAnalyzerAssemblyLoader loader, Assembly assembly)
+            {
+                var undidHook = false;
+                try
+                {
+                    // Have to unhook resolve here otherwise calls to Load will hit the resolver and 
+                    // we will end up bringing these assemblies into the Load context by handling them
+                    // there.
+                    undidHook = loader.EnsureResolvedUnhooked();
+                    var name = assembly.FullName;
+                    var other = AppDomain.CurrentDomain.Load(name);
+                    return other is null || other != assembly;
+                }
+                catch
+                {
+                    return true;
```

Comment:
Can you help me understand this bit? What in the above try block is going to throw and why does that make us think the assembly is in the LoadFrom context?

Reply:
A call to `AppDomain.Load(string)` will only succeed if one of the following is true: 

1. An `Assembly` with that name is already present in the _Load_ context
2. An `Assembly` with that name was found in the probing path of the `AppDomain`. Generally this is the directory where the application was run from or assemblies in the GAC
4. The `AppDomain.ResolveAssembly` event was able to find an assembly with the specified name. 

When none of these are true an exception is thrown and that demonstrates it's in the _LoadFrom_ context not _Load_. 

Next I expect you'll want to know why I also test for `null`. Mostly for completeness because I can't prove to myself that `null` won't be returned in edge cases :smile: